exporter/logicmonitorexporter/logs_exporter_test.go

```diff
@@ -0,0 +1,167 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logicmonitorexporter
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+func Test_NewLogsExporter(t *testing.T) {
+
+	type args struct {
+		config *Config
+		logger *zap.Logger
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			"Create NewLogExporter",
+			args{
+				config: &Config{
+					ExporterSettings: config.NewExporterSettings(component.NewID("logicmonitor")),
+					HTTPClientSettings: confighttp.HTTPClientSettings{
+						Endpoint: "http://example.logicmonitor.com/rest",
+					},
+					APIToken: APIToken{AccessID: "testid", AccessKey: "testkey"},
+				},
+				logger: zap.NewNop(),
+			},
+			false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			set := componenttest.NewNopExporterCreateSettings()
+			le := newLogsExporter(tt.args.config, set)
+			if (le == nil) != tt.wantErr {
+				t.Errorf("newLogsExporter() error in creating newLogsExporter, wantErr %v", tt.wantErr)
+				return
+			}
+		})
+	}
+}
+
+type Response struct {
+	Success bool   `json:"success"`
+	Message string `json:"message"`
+}
+
+func TestPushLogData(t *testing.T) {
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		response := Response{
+			Success: true,
+			Message: "Logs exported successfully",
+		}
+		body, _ := json.Marshal(response)
+		_, _ = w.Write(body)
+	}))
+
+	type args struct {
+		ctx context.Context
+		lg  plog.Logs
+	}
+
+	type fields struct {
+		config *Config
+		logger *zap.Logger
+	}
+
+	cfg := &Config{
+		HTTPClientSettings: confighttp.HTTPClientSettings{
+			Endpoint: ts.URL,
+		},
+		APIToken: APIToken{AccessID: "testid", AccessKey: "testkey"},
+	}
+
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+	}{
+		{
+			name: "Send Log data",
+			fields: fields{
+				logger: zap.NewNop(),
+				config: cfg,
+			},
+			args: args{
+				ctx: context.Background(),
+				lg:  createLogData(1),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Setenv("LOGICMONITOR_ACCOUNT", "localdev")
```

Comment:
Is it possible to not use environment variables here? Could this be set programatically?

Using environment variables means that these tests can't run in parallel and it makes it harder to validate across different tests.

Reply:
Yes, handled programatically.